### ISSUE TITLE
BUG: 4.18-beta - wrong stats displayed after error

### ISSUE THREAD

**Issue by bcurran3** [RandomEngy/VidCoder#435]
This was reported before in a previous build, but I can't find the issue # at the moment. Will edited if I can find later.

![image](https://user-images.githubusercontent.com/14026600/47376517-a4f1e980-d6a7-11e8-8627-251d7e030378.png)

**Comment by RandomEngy** [comment]
Yes, I added logging to help identify what's going wrong. Please share the application log on pastebin.

**Comment by RandomEngy** [comment]
Great, but the logs would still be helpful to fix this.

**Comment by RandomEngy** [comment]
Thanks for the logs. From the log messages and examining the code I couldn't see any obvious way for the problem to arise. I did notice that the code might be subject to some threading/race conditions so I added some extra locking in 4.21 Beta. Please let me know if it still happens in that version.

**Comment by bcurran3** [comment]
Related to: https://github.com/RandomEngy/VidCoder/issues/402
